File: JavaLibrary/src/main/java/com/flat/simplex/parser/logic/Context.java
package com.flat.simplex.parser.logic;

import com.flat.simplex.lexer.Token;
import com.flat.simplex.parser.logic.error.Error;

import java.util.ArrayList;
import java.util.List;

public class Context extends Block {

    private final ArrayList<Error> errors = new ArrayList<>();
    
    public Context(Token token) {
        super(null, token);
    }

    @Override
    public Field getField(String fieldName) {
        return fields.get(fieldName);
    }

    public List<String> getAllFieldNames() {
        return fields.keySet().stream().toList();
    }

    @Override
    public void error(Token token, String description) {
        errors.add(new Error(Error.Type.Syntax, description, token, token));
    }
    
    @Override
    public void warning(Token token, String description) {
        errors.add(new Error(Error.Type.Warning, description, token, token));
    }
    
    @Override
    public void addError(Error error) {
        errors.add(error);
    }
    
    @Override
    public ArrayList<Error> getErrors() {
        return errors;
    }
}

File: SimplexApp/simplex/parser/logic/Context.js
var Error;
const Block = require("simplex/parser/logic/Block.js");

class Context extends Block {
    constructor(token) {
        super(null, token);

        this.errors = [];
    }

    getField(fieldName) {
        return this.fields.get(fieldName);
    }

    getAllFieldNames() {
        return Array.from(this.fields.keys());
    }

    error(token, description) {
        this.errors.push(new Error(Error.Syntax, description, token, token));
    }

    warning(token, description) {
        this.errors.push(new Error(Error.Warning, description, token, token));
    }

    addError(error) {
        this.errors.push(this.error);
    }

    getErrors() {
        return this.errors;
    }
}

module.exports = Context;
Error = require("simplex/parser/logic/error/Error.js");